### ISSUE TITLE
DDL Download Issue.

### ISSUE THREAD

**Issue by Obebudda** [evilhero/mylar#2366]
Hi, whenever I try to download anything for The Flash issue 76 and below the below error is thrown. I collected the log with Debugging turned on. Any chance someone can tell me what I am doing wrong? Mylar information is below as well as the log. Only search providers being used is DDL.

Mylar Version: master
-- git build 5ffe8fbd3080e11884c6d11ad3aacbc9f6a73cb3.
Python Version : 2.7.15
Language : en_US.cp1252

1-Nov-2019 02:47:31 - DEBUG   :: mylar.queueissue.1341 : Thread-14 : ComicID: 91750
01-Nov-2019 02:47:31 - DEBUG   :: mylar.queueissue.1342 : Thread-14 : mode: want
01-Nov-2019 02:47:31 - INFO    :: mylar.queueissue.1432 : Thread-14 : Initiating manual search for The Flash issue: 76
01-Nov-2019 02:47:31 - INFO    :: mylar.queueissue.1475 : Thread-14 : [WANTED-SEARCH] Now Queueing The Flash (2016) #76 for search
01-Nov-2019 02:47:33 - DEBUG   :: mylar.dbUpdate.85 : Thread-15 : Refresh sequence set to fire every 5 minutes for 7 day(s)
01-Nov-2019 02:47:33 - DEBUG   :: mylar.job_management.3576 : Thread-15 : ReScheduled job: DB Updater to 2019-11-01 06:52:33
01-Nov-2019 02:47:33 - INFO    :: mylar.dbUpdate.353 : Thread-15 : Update complete
01-Nov-2019 02:47:36 - INFO    :: mylar.search_queue.3145 : SEARCH-QUEUE : [SEARCH-QUEUE] Now loading item from search queue: {'comicname': u'The Flash', 'issueid': u'716316', 'seriesyear': u'2016', 'booktype': u'Print', 'issuenumber': u'76', 'comicid': u'91750'}
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.63 : SEARCH-QUEUE : Publisher is : DC Comics
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.71 : SEARCH-QUEUE : Issue Title given as : death and the speed force part one
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.102 : SEARCH-QUEUE : Checking for torrent enabled.
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.153 : SEARCH-QUEUE : nzbprovider(s): []
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.156 : SEARCH-QUEUE : There are 0 torrent providers you have selected.
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.161 : SEARCH-QUEUE : There are : 0 nzb providers you have selected
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.166 : SEARCH-QUEUE : there are 1 Direct Download providers that are currently enabled.
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.180 : SEARCH-QUEUE : search provider order is ['DDL']
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.247 : SEARCH-QUEUE : calpha/cnumber: None / None
01-Nov-2019 02:47:36 - DEBUG   :: mylar.search_init.283 : SEARCH-QUEUE : Initiating Search via : api
01-Nov-2019 02:47:36 - INFO    :: mylar.NZB_SEARCH.513 : SEARCH-QUEUE : Shhh be very quiet...I'm looking for The Flash issue: 76 (2019) using ddl.
01-Nov-2019 02:47:36 - DEBUG   :: mylar.NZB_SEARCH.577 : SEARCH-QUEUE : findloop: 0 / findcount: 1
01-Nov-2019 02:47:36 - DEBUG   :: mylar.NZB_SEARCH.623 : SEARCH-QUEUE : Sending request to DDL site for : The Flash 76
01-Nov-2019 02:47:37 - INFO    :: mylar.search_results.89 : SEARCH-QUEUE : There are 12 results
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : The Flash #76 (2019) [35 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : Flash Forward #2 (2019) [37 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : Batman #76 (2019) [21 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : The Flash #71 (2019) [30 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : Flashpoint (Story Arc) (2010-2011) [2.04 G]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : The Flash #56 (2018) [33 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : The Flash #54 (2018) [35 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : The Flash #52 (2018) [33 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : Secret Origins of the Super DC Heroes (1976) [90 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : Convergence - Flashpoint Book 1 - 2 (2015) [724 M]
01-Nov-2019 02:47:37 - DEBUG   :: mylar.search_results.161 : SEARCH-QUEUE : Flashpoint - The World of Flashpoint Featuring Green Lantern (2012) [305 M]
01-Nov-2019 02:47:37 - ERROR   :: mylar.excepthook.315 : SEARCH-QUEUE : Uncaught exception: Traceback (most recent call last):
  File "c:\Mylar\mylar\logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "c:\Python27\lib\threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "c:\Mylar\mylar\helpers.py", line 3147, in search_queue
    ss_queue = mylar.search.searchforissue(item['issueid'])
  File "c:\Mylar\mylar\search.py", line 1855, in searchforissue
    foundNZB, prov = search_init(ComicName, IssueNumber, str(IssueYear), SeriesYear, Publisher, IssueDate, StoreDate, actissueid, AlternateSearch, UseFuzzy, ComicVersion, SARC=SARC, IssueArcID=IssueArcID, mode=mode, rsscheck=rsscheck, ComicID=ComicID, filesafe=Comicname_filesafe, allow_packs=allow_packs, oneoff=oneoff, manual=manual, torrentid_32p=TorrentID_32p, digitaldate=DigitalDate, booktype=booktype)
  File "c:\Mylar\mylar\search.py", line 359, in search_init
    findit = NZB_SEARCH(ComicName, IssueNumber, ComicYear, SeriesYear, Publisher, IssueDate, StoreDate, searchprov, send_prov_count, IssDateFix, IssueID, UseFuzzy, newznab_host, ComicVersion=ComicVersion, SARC=SARC, IssueArcID=IssueArcID, RSS="no", ComicID=ComicID, issuetitle=issuetitle, unaltered_ComicName=unaltered_ComicName, allow_packs=allow_packs, oneoff=oneoff, cmloopit=cmloopit, manual=manual, torznab_host=torznab_host, torrentid_32p=torrentid_32p, digitaldate=digitaldate, booktype=booktype)
  File "c:\Mylar\mylar\search.py", line 625, in NZB_SEARCH
    bb = b.search()
  File "c:\Mylar\mylar\getcomics.py", line 68, in search
    return self.search_results()
  File "c:\Mylar\mylar\getcomics.py", line 128, in search_results
    option_find = option_find.findNext(text=True)
AttributeError: 'NoneType' object has no attribute 'findNext'

**Comment by evilhero** [comment]
It's a problem with the web parser for getcomics - I'll have to code dive into it and see what's going on.

**Comment by Obebudda** [comment]
Okay thanks for the update.

**Comment by rutskiuk** [comment]
Any news on this one? I'd noticed a while back that DDL wasn't working. Sources are clearly available so the parser seems to be causing me a bit of a backlog.

**Comment by mylar3** [comment]
Since we're no longer patching non-critical bugs - best advice would be to migrate to the python3 fork located [here](https://github.com/mylar3/mylar3) since we've moved all our dev to that.

Alot of the bugs that were causing issues on the py2 branch have already been resolved awhile ago on the new repo. There are still some that we're working thru obviously, but it's very usable as a daily driver.

Easy to migrate over as well - just git clone the new repo, and then copy your existing mylar.db, config.ini and cache folder into the new repo location. Edit your config.ini to make sure that any references to your old Mylar location are updated to the new location prior to starting it up (ie. logs_dir, cache_dir, git_branch, git_user, etc)

**Comment by rutskiuk** [comment]
I moved over to the PY3 Branch after eventually reading the bit about Py 3.8 not being supported for windows (wish I'd read that bit sooner)
Now I'm on the new branch I'm still seeing no DDL downloads. I enabled verbose logging and did a search for an example of an issue available for download:

2020-05-07 00:33:32 | INFO | Could not find Issue 3 of Batman: The Adventures Continue (2020) using ddl [api]

2020-05-07 00:42:24 | DEBUG | [SPECIAL-CHARACTER ISSUE] Possible issue # : –
-- | -- | --
2020-05-07 00:42:24 | DEBUG | NEWLY SPLIT REORGD: ['Batman', 'XCV', 'The', 'Adventures', 'Continue', '#3', '(2020)']
2020-05-07 00:42:24 | DEBUG | Entry: Batman – The Adventures Continue #3 (2020)

I think it can see the issue but doesn't like the naming structure but by the looks of it this is the naming structure used by GetComics. Am I misunderstanding the logs here?
